client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java

```diff
@@ -56,6 +56,24 @@
  */
 public abstract class AbstractSender extends AbstractHonoClient implements MessageSender {
 
+    /**
+     * Client facing error message for the case of a message sender not receiving a delivery update in time.
+     * May be used as {@link ServerErrorException#clientFacingMessage}.
+     */
+    public static final String ERROR_SEND_MESSAGE_TIMEOUT = "consumer failed to indicate in time whether it has processed the message";
```

Comment:
FMPOV you have now opened Pandora's Box of (localized) messages ;-)
I do not think that we should populate our constants classes with full fledged error messages ...
Instead, we should probably use resource bundles which have been designed for this purpose.

Reply:
Using a resource bundle just for these error messages seems a bit overkill to me. We also don't use a resource bundle for the `ClientErrorException` messages already being included in HTTP responses.
I mainly introduced these constants to match the error message in the unit tests. As we rarely use such error message checks in unit tests, I could just remove them and then also remove the constants.

Reply:
FMPOV it would actually be the right choice for the problem at hand. I also do not think that it is much effort to implement.

Reply:
So you mean introducing a resource bundle, used then also for the `ClientErrorException` messages?

Reply:
You mean for the already existing messages? IMHO that should be the goal then, yes. We could maybe use a step-by-step approach here but in the end we should use the same mechanism. FMPOV this is even more desirable if we want to provide more elaborate messages than *bad JSON* ...